### DISCUSSION TITLE
Product Specific Shipping

### DISCUSSION THREAD

**Discussion by simon-online** [saleor/saleor#11857] [Q&A] [answered]
We are building an ecommerce store that requires a mix of weight based rule shipping costs and product specific shipping costs ie. a fixed price per product item shipping cost. For example:

- product A to shipping zone 1 has a shipping cost of $10 per item
- product B to shipping zone 1 has a shipping cost of $5 per item
- the additional shipping cost for all other order items is calculated using the weight based shipping rules for shipping zone 1

The product specific shipping costs are typically used for oversized / non-standard products which get shipped separately from the rest of the order.

Using Saleor I'd say we would need to build a custom shipping calculation and make use of metadata fields for the products with specific shipping pricing.

Is this the best approach to achieve this using Saleor?

**Comment by timuric** [comment, marked as answer]
@simon-online indeed, you would need to use a webhook for that https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/shipping-webhooks#subscribe-to-the-webhook-1
